Psr/Log/LoggerAwareTrait.php

```diff
@@ -12,7 +12,7 @@ trait LoggerAwareTrait
      *
      * @var LoggerInterface|null
      */
-    protected $logger;
+    protected ?LoggerInterface $logger;
```

Comment:
It should be initialized with null value, isn't it?

```suggestion
    protected ?LoggerInterface $logger = null;
```

Reply:
null is implied as initial value

Reply:
Not in case when property types used

Reply:
@andrew-demb agree, this should be initialized as `null`.

Reply:
In general, I would argue that it should be as is.  Or even just `LoggerInterface` on its own.  Possible null properties are a bug that makes all code that uses it harder, and if someone gave me this code today I'd tell them they were doing it wrong.

The spec doesn't specify anything about the property specifically.  However, it's been documented in the implementation here as `LoggerInterface|null` for 8 years, and *technically* uninitialized is not the same, so not providing a default would add another possible type over what it was previously.  So...

All that is to say I think allowing null here is a mistake to begin with, but allowing null and defaulting to null is the closest to the original behavior, so we'll go with that anyway even though it's the less-good design.  Will update.